evalml/pipelines/components/transformers/imputers/knn_imputer.py

```diff
@@ -51,16 +50,27 @@ def fit(self, X, y=None):
 
         """
         X = infer_feature_types(X)
-        X, _ = drop_natural_language_columns(X)
 
         nan_ratio = X.isna().sum() / X.shape[0]
+
+        # Keep track of the different types of data in X
         self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
+        self._natural_language_cols = list(
+            X.ww.select("NaturalLanguage", return_schema=True).columns.keys(),
+        )
+
+        # Only impute data that is not natural language columns or fully null
+        self._cols_to_impute = [
```

Comment:
This change to use a `_cols_to_impute` variable to manage the fully null and natural language column logic here and at the simple imputer was a change that, to me, makes the imputer logic easier to understand. It makes the handling more explicit rather than a sequence of events to keep track of. It should also have slightly more efficient woodwork usage.

One implication of this is that we never even pass fully null columns to the imputer component_obj, which could be problematic for test data that may not be fully null, but in that case, we'd have no information from the training data to use to impute data, and in the previous call to `pd.DataFrame(X_t, columns=not_all_null_or_natural_language_cols)`, it assumes the fully null columns wouldn't be present anyway, so I'm assuming that it's okay to not pass the fully null columns in at all.

Reply:
It should be ok as long as we're not passing the fully null or natural language column as input for both fit and transform. It is also ok to pass the full null columns in - I believe the sklearn imputer's drop them automatically.

Reply:
![Screenshot 2023-03-02 at 2 05 39 PM](https://user-images.githubusercontent.com/15271480/222527075-40543a74-c5b6-4443-a3d9-b2c595d39ec5.png)

Reply:
yep! It simplified things a bit to not pass them in at either fit or transform, but agreed it's something we need to be consistent on.